File: src/main/java/uk/co/threebugs/mochiwhattotrade3/setupgroups/SetupGroupsMapper.java
package uk.co.threebugs.mochiwhattotrade3.setupgroups;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.nio.file.Path;
import java.nio.file.Paths;

@Mapper(componentModel = "spring")
public interface SetupGroupsMapper {
    SetupGroupsDto toDto(SetupGroups setupGroups);

    SetupGroups toEntity(SetupGroupsDto setupGroupsDto);

    @Mapping(target = "id", ignore = true)
    SetupGroups toEntity(SetupGroupsFile setupGroupsDto);

    default String pathToString(Path path) {
        return path == null ? null : path.toString();
    }

    default Path stringToPath(String path) {
        return path == null ? null : Paths.get(path);
    }

}

File: backend/src/main/kotlin/uk/co/threebugs/darwinexclient/setupgroups/SetupGroupsMapper.kt
package uk.co.threebugs.darwinexclient.setupgroups

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import java.nio.file.Path

@Mapper(componentModel = "spring")
abstract class SetupGroupsMapper {
    abstract fun toDto(setupGroups: SetupGroups): SetupGroupsDto
    abstract fun toEntity(setupGroupsDto: SetupGroupsDto): SetupGroups

    @Mapping(target = "id", ignore = true)
    abstract fun toEntity(setupGroupsDto: SetupGroupsFile): SetupGroups
    fun toString(path: Path?): String? {
        return path?.toString()
    }

    fun toPath(path: String?): Path? {
        return path?.let { Path.of(it) }
    }
}
